buildbot/configure.py

```diff
@@ -198,6 +198,12 @@ def do_configure(args):
             "-DSYCL_LIBCXX_INCLUDE_PATH={}".format(args.libcxx_include),
             "-DSYCL_LIBCXX_LIBRARY_PATH={}".format(args.libcxx_library)])
 
+    if 'openmp' in llvm_external_projects:
+        omp_dir = os.path.join(abs_src_dir, "openmp")
```

Comment:
@bader, yes. I am building "in-tree" openmp. 

> we will need a new option as there is no existing `configure.py`

What do you think about the following changes?

```diff
$ git diff buildbot/configure.py
diff --git a/buildbot/configure.py b/buildbot/configure.py
index 4fd70a211187..af5eec9205cf 100644
--- a/buildbot/configure.py
+++ b/buildbot/configure.py
@@ -31,6 +31,10 @@ def do_configure(args):
     if args.llvm_external_projects:
         llvm_external_projects += ";" + args.llvm_external_projects.replace(",", ";")
 
+    llvm_in_tree_projects = []
+    if args.enable_additional_in_tree_projects:
+        llvm_in_tree_projects = ";" + args.enable_additional_in_tree_projects.replace(",", ";")
+
     llvm_dir = os.path.join(abs_src_dir, "llvm")
     sycl_dir = os.path.join(abs_src_dir, "sycl")
     spirv_dir = os.path.join(abs_src_dir, "llvm-spirv")
@@ -39,7 +43,7 @@ def do_configure(args):
     libdevice_dir = os.path.join(abs_src_dir, "libdevice")
     fusion_dir = os.path.join(abs_src_dir, "sycl-fusion")
     llvm_targets_to_build = args.host_target
-    llvm_enable_projects = 'clang;' + llvm_external_projects
+    llvm_enable_projects = 'clang;' + llvm_external_projects + llvm_in_tree_projects
     libclc_targets_to_build = ''
     libclc_gen_remangled_variants = 'OFF'
     sycl_build_pi_hip_platform = 'AMD'
@@ -198,10 +202,8 @@ def do_configure(args):
             "-DSYCL_LIBCXX_INCLUDE_PATH={}".format(args.libcxx_include),
             "-DSYCL_LIBCXX_LIBRARY_PATH={}".format(args.libcxx_library)])
 
-    if 'openmp' in llvm_external_projects:
-        omp_dir = os.path.join(abs_src_dir, "openmp")
-        cmake_cmd.extend([
-            "-DLLVM_EXTERNAL_OPENMP_SOURCE_DIR={}".format(omp_dir),
+    if 'openmp' in llvm_in_tree_projects:
+      cmake_cmd.extend([
             "-DOPENMP_ENABLE_LIBOMPTARGET=OFF"])
 
     print("[Cmake Command]: {}".format(" ".join(cmake_cmd)))
@@ -254,6 +256,8 @@ def main():
     parser.add_argument("--libcxx-library", metavar="LIBCXX_LIBRARY_PATH", help="libcxx library path")
     parser.add_argument("--use-lld", action="store_true", help="Use LLD linker for build")
     parser.add_argument("--llvm-external-projects", help="Add external projects to build. Add as comma seperated list.")
+    parser.add_argument("--enable-additional-in-tree-projects",
+                        help="Add additional in-tree projects to build. Add as comma seperated list.")
     parser.add_argument("--ci-defaults", action="store_true", help="Enable default CI parameters")
     parser.add_argument("--enable-plugin", action='append', help="Enable SYCL plugin")
     parser.add_argument("--disable-fusion", action="store_true", help="Disable the kernel fusion JIT compiler")
```

Reply:
I'm fine with the existing patch and the approach proposed the comment. I don't have preferences here. @aelovikov-intel, @intel/dpcpp-devops-reviewers, do you have any preferences for configuring OpenMP project?

Reply:
I'm slightly in favor of the change suggested in the comment above, with a few modifications:
1) Name the option "--llvm-projects"
2) 
```
+ llvm_in_tree_projects = [] + if args.enable_additional_in_tree_projects:
+ llvm_in_tree_projects = ";" + args.enable_additional_in_tree_projects.replace(",", ";")
```
that looks wrong - the same variable is used both as a list and as a string in different places.

Reply:
@aelovikov-intel, agree. I made the appropriate changes.

https://github.com/ravil-mobile/intel-llvm/blob/e2e889197cf1a73983d16bc9d2e591481d28d709/buildbot/configure.py#L34-L36

Reply:
Do we need to differentiate between "external" and "internal" projects? From what I can tell, it goes into the same CMake variable, and we don't handle them differently here. So, I like @aelovikov-intel's proposal to just have a single `--llvm-projects` instead of those two.

Reply:
External projects are handled differently.

**LLVM_EXTERNAL_PROJECTS:STRING**
Semicolon-separated list of additional external projects to build as part of llvm. For each project LLVM_EXTERNAL_<NAME>_SOURCE_DIR have to be specified with the path for the source code of the project. Example: -DLLVM_EXTERNAL_PROJECTS="Foo;Bar" -DLLVM_EXTERNAL_FOO_SOURCE_DIR=/src/foo -DLLVM_EXTERNAL_BAR_SOURCE_DIR=/src/bar.